### DISCUSSION TITLE
Declare optional execution priority on test cases/classes

### DISCUSSION THREAD

**Discussion by bert2** [xunit/xunit#2524] [Ideas]
The idea is that some test cases or test classes can be marked to have a lower execution priority. This causes them to be executed later than other tests which by default have a higher execution priority.

I find this useful for huge collections of tests where some take a lot more time than others. For instance, because they setup an in-memory database.

With execution priority I could simply run all tests at once and have faster feedback from the majority of the (fast-running) tests.

Note that this is not intended to model interdependencies between test. I'm aware that the xunit community regards this as a bad practice.

**Comment by bradwilson** [comment]
This is an interesting idea but outside the scope of a pure unit test usage, which tends to be our focus. Our extensibility points would allow extensions that could prioritize the running of things in specific orders; a [simple version exists as a sample today](https://github.com/xunit/samples.xunit/tree/main/TestOrderExamples). More comprehensive control of the execution pipeline, if needed, is available as well; the [Observation sample](https://github.com/xunit/samples.xunit/tree/main/ObservationExample) shows how to take full control over the execution pipeline.
